src/test/java/com/devonfw/sample/archunit/ComponentRuleTest.java

```diff
@@ -0,0 +1,328 @@
+package com.devonfw.sample.archunit;
+
+import static com.tngtech.archunit.lang.syntax.ArchRuleDefinition.noClasses;
+
+import com.tngtech.archunit.core.domain.Dependency;
+import com.tngtech.archunit.core.domain.JavaClass;
+import com.tngtech.archunit.core.importer.ImportOption;
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchCondition;
+import com.tngtech.archunit.lang.ArchRule;
+import com.tngtech.archunit.lang.ConditionEvents;
+import com.tngtech.archunit.lang.SimpleConditionEvent;
+
+@AnalyzeClasses(packages = "com.devonfw.sample.archunit", importOptions = ImportOption.DoNotIncludeTests.class)
+public class ComponentRuleTest {
+
+  public static final ArchCondition<JavaClass> haveComponentServiceLayerDependingOnDiffComponentsServiceLayer = new ArchCondition<JavaClass>(
+      "have dependencies, towards another components service layer (Rule-C3).") {
```

Comment:
Using `as("message")` enforces printing only the `message `string.
Because concatenates the fluent API messages.
E.g.:

```
noClasses()
      .that(DescribedPredicate("message1"))
      .should(ArchCondition("message2"))
      .because("message3")
```


would result in _"no classes that message1 should message2, because message3"_.

I reworked my implementation and utilize now DescribedPredicates and ArchConditions.
There are more appropriate messages inside DescribedPredicates and ArchConditions now (which help to understand the code better). Nevertheless, I use `as("message")` to return a shorter error message.